### PR TITLE
[Peer Review] developer getting started experience

### PR DESCRIPTION
**Pull request by tpoitras**
This isn't totally ready for tech review or peer review yet.

I'm still tweaking, have changed the sample app to a nodejs-mongodb template, but haven't tested everything yet, as the stage console is down today.

### CODE REVIEW

**Comment by tpoitras** [comment]
OK, the Developer (Web Console) part is done. There is a video, but I cannot figure out how to embed it. The asciidoctor user guide describes how to embed a local mp4, as well as a youtube clip. I've tried both but it doesn't work for me. 

This is the YouTube clip: https://www.youtube.com/watch?v=rF5z_SjXx9U

File: getting_started/developers/developers_cli.adoc [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+// end::overview[]
+
+// tag::beforeyoubegin[]
+
+== Before You Begin
+
+Before you can get started using OpenShift Enterprise:
+
+- You must be able to access a running instance of OpenShift Enterprise. If you do not have access, then contact your system administrator.
+- Your instance must be pre-configured by a system administrator with sample templates and builder images.
```
Comment (tpoitras) [inline getting_started/developers/developers_cli.adoc]: @adellape -- do you know how I can point to the correct templates and builder images that are in use on the stage console?

**Comment by tpoitras** [comment]
OK, Sean Huck has the video hosted on the portal now, and I've embedded it into the docs here. It works on a local build, and when I push that build to my file.bne

I think this is safe to push now. For some reason though, on my machine, it only works on Chrome, and not Firefox.

**Comment by adellape** [comment]
@tpoitras @vikram-redhat Sorry, late to the party on this, but some broken-link issues introduced here that should get addressed in a follow-up (and apologies if these have already been discussed and/or are being addressed already):

1) Per:

> the Developer (CLI) part is to be continued. If this merges, I'll finish it in another PR.

If the CLI topic is still WIP, in https://docs.openshift.com/enterprise/3.0/getting_started/overview.html (and the Origin version https://docs.openshift.org/latest/getting_started/overview.html) there is currently a link to `getting_started/developers/developers_cli.html` that 404s since it's omitted from the `build_cfg`.

2) In:

(OSE) https://docs.openshift.com/enterprise/3.0/welcome/index.html
(Origin) https://docs.openshift.org/latest/welcome/index.html

There is a link to `getting_started/developers.html` that 404s now since it was removed in this PR.

3) `developers_console.html` is not conditionalized for Origin vs OSE, so in the Origin build (https://docs.openshift.org/latest/getting_started/developers/developers_console.html) it references "OpenShift Enterprise" a couple times.

4) Also in the Origin build:

https://docs.openshift.org/latest/getting_started/overview.html

For "Platform administrators" it links to the Quick Install:

https://docs.openshift.org/latest/getting_started/overview.html

Which for Origin just links over to the existing Getting Started for Administrators (which is valid for Origin):

https://docs.openshift.org/latest/getting_started/administrators.html

Could just link directly to that instead (currently being expanded in https://github.com/openshift/openshift-docs/pull/959 as well).

5) Two spots in `developers_console.html` are rendering example links that go nowhere: 

```
Click Edit Parameters, then update the source repository URL to your forked repository: https://github.com/<your_github_username>/nodejs-ex.git.
```

and

```
In the web console, view the overview page to determine the web address for the application. For example: http://nodejs-mongodb-example.openshiftapps.com
```

I'm pretty sure I've gotten this to behave correctly elsewhere (i.e., do not render as a link), but off the top of my head can't remember what works best. Maybe just italics, or maybe the `$` or `+` passthrough macros in http://www.methods.co.nz/asciidoc/userguide.html#X77 ?

**Comment by tpoitras** [comment]
Alex, you legend! Thanks for finding these. I have addressed all of them and submitted PR#962:

https://github.com/openshift/openshift-docs/pull/962

**Pull request merged**